crates/net/network/src/peers/manager.rs

```diff
@@ -458,6 +473,24 @@ impl Default for PeersConfig {
                 max_inbound: 30,
             },
             reputation_weights: Default::default(),
+            ban_list: Default::default(),
         }
     }
 }
+
+/// Configuration for the automatic removal of unwanted peers
+#[derive(Debug, Default)]
+pub struct BanList {
+    /// blacklisted peer ids
+    blacklisted_peers: HashSet<PeerId>,
+    /// blacklisted ips
+    blacklisted_ips: HashSet<IpAddr>,
+}
+
+impl BanList {
+    /// checks the ban list for the given ip address and peer_id and returns true
+    /// if the address is in the ban list
+    pub(self) fn is_banned(&self, ip: &IpAddr, peer_id: &PeerId) -> bool {
```

Comment:
We probably also want a simple is_ip_banned and is_peerid_banned for future library users that may know one but not the other, but lgtm here